### PR TITLE
JAVA-2682: Don't retry writes unless all writes in the list are retryable

### PR DESCRIPTION
**Pull request by jyemin**
The fix is easy but I'm not too happy with just having a unit test of BulkWriteBatch.  Is there a straightforward way to write an integration test for this?

https://evergreen.mongodb.com/version/5a1f36b0e3c3315c9101b860

### CODE REVIEW

File: driver-core/src/test/unit/com/mongodb/operation/BulkWriteBatchSpecification.groovy [hunk 1]
```diff
@@ -41,8 +44,13 @@ import static com.mongodb.connection.ServerConnectionState.CONNECTED
 
 class BulkWriteBatchSpecification extends Specification {
     def namespace = new MongoNamespace('db.coll')
-    def serverDescription = ServerDescription.builder().address(new ServerAddress()).state(CONNECTED).build()
-    def connectionDescription = new ConnectionDescription(new ServerId(new ClusterId(), serverDescription.getAddress()))
+    def serverDescription = ServerDescription.builder().address(new ServerAddress()).state(CONNECTED)
```
Comment (rozza) [inline driver-core/src/test/unit/com/mongodb/operation/BulkWriteBatchSpecification.groovy]: Why are these changes needed?

**Comment by rozza** [comment]
Integration wise - you could set up a single failure for a bulk write and ensure it throws when retry writes is true.

**Comment by rozza** [comment]
Thinking about it, there should be a spec test to cover this scenario.

File: driver-core/src/test/unit/com/mongodb/operation/BulkWriteBatchSpecification.groovy (hunk 1 above)
Comment (jyemin) [reply driver-core/src/test/unit/com/mongodb/operation/BulkWriteBatchSpecification.groovy]: Ah, good question.  I should have commented on that. Without this change OperationHelper#canRetryWrite returns false and the test passes even if the code change is not made.  I almost missed that because I changed the code before writing the test, but just to be sure I reverted the code change and re-ran the test and was surprised to see that it still passed.  A good less for me to do TDD religiously even for small changes like this.

**Comment by jyemin** [comment]
Yeah, agreed.  It's an easy test to write, as the server reports an error if the driver sends a txnNumber on a delete/update command that contains a write that can update more than one document.

**Comment by jyemin** [comment]
Added an integration test.  I'll open a SPEC ticket for a new YAML test
